Code/Framework/AzFramework/Tests/DocumentPropertyEditor/CvarAdapterTests.cpp

```diff
@@ -0,0 +1,170 @@
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+#include <Tests/DocumentPropertyEditor/DocumentPropertyEditorFixture.h>
+
+namespace AZ::DocumentPropertyEditor::Tests
+{
+    AZ_CVAR(int32_t, dpe_TestInt, 1, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test int32");
+    AZ_CVAR(uint8_t, dpe_TestUint, 22, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test uint8");
+    AZ_CVAR(double, dpe_TestDbl, 4.0, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test double");
+    AZ_CVAR(CVarFixedString, dpe_TestString, "test", nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test string");
+    AZ_CVAR(bool, dpe_TestBool, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test bool");
+    AZ_CVAR(AZ::Vector2, dpe_TestVec2, AZ::Vector2::CreateZero(), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test vec2");
+    AZ_CVAR(AZ::Vector3, dpe_TestVec3, AZ::Vector3::CreateOne(), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test vec3");
+    AZ_CVAR(AZ::Color, dpe_TestColor, AZ::Color(), nullptr, ConsoleFunctorFlags::DontReplicate, "DPE test color");
+
+    class CvarAdapterDpeTests : public DocumentPropertyEditorTestFixture
+    {
+    public:
+        void SetUp() override
+        {
+            DocumentPropertyEditorTestFixture::SetUp();
+            m_console = AZStd::make_unique<AZ::Console>();
+            m_console->LinkDeferredFunctors(AZ::ConsoleFunctorBase::GetDeferredHead());
+            AZ::Interface<AZ::IConsole>::Register(m_console.get());
+            m_adapter = AZStd::make_unique<CvarAdapter>();
+        }
+
+        void TearDown() override
+        {
+            m_adapter.reset();
+            AZ::Interface<AZ::IConsole>::Unregister(m_console.get());
+            m_console.reset();
+            DocumentPropertyEditorTestFixture::TearDown();
+        }
+
+        Dom::Value GetEntryRow(AZStd::string_view cvarName)
+        {
+            Dom::Value rows = m_adapter->GetContents();
+            for (auto it = rows.ArrayBegin(); it != rows.ArrayEnd(); ++it)
+            {
+                Dom::Value label = (*it)[0].GetNodeValue();
+                if (label.GetString() == cvarName)
+                {
+                    return *it;
+                }
+            }
+            return {};
+        }
+
+        Dom::Value GetEntryValue(AZStd::string_view cvarName)
+        {
+            Dom::Value row = GetEntryRow(cvarName);
+            EXPECT_FALSE(row.IsNull());
+            return row[1].GetNodeValue();
+        }
+
+        void SetEntryValue(AZStd::string_view cvarName, Dom::Value value)
+        {
+            Dom::Value row = GetEntryRow(cvarName);
+            ASSERT_FALSE(row.IsNull());
+            auto result = Nodes::PropertyEditor::OnChanged.InvokeOnDomNode(row[1], value);
+            EXPECT_TRUE(result.IsSuccess());
+            if (!result.IsSuccess())
+            {
+                EXPECT_EQ("", result.GetError());
```

Comment:
Seems like an odd thing to validate here. If the result is a failure, we are expecting the error to be empty? Can you explain why we need to validate this here?

Reply:
This is just a quick and dirty way to spit out the error message in the log if this ever fails, I can use an AZ_Warning instead.

Reply:
Updated to use a clearer `AZ_Error`